libparsec/crates/serialization_format/tests/common/libparsec_types_mock.rs

```diff
@@ -11,8 +11,14 @@ pub type Integer = i64;
 #[derive(serde::Deserialize, serde::Serialize, Clone, Debug, PartialEq, Eq)]
 pub struct DeviceID(pub String);
 
+#[allow(unused)]
+pub enum ProtocolFamily {
+    Family,
```

Comment:
Why is this mock needed ?

Reply:
It's used for testing the crate itself

Reply:
Yes, the `libparsec_serialization_format` crate generates in its proc macros some code that makes uses of the types in `libparsec_types`.
So that means `libparsec_serialization_format`'s tests requires `libparsec_types` as dependencies, but this is cumbersome (don't remember the specifics, but I guess it has to do with circular dependencies).
So instead of directly relying on `libparsec_types`, the generated code refers to something like `super::libparsec_types`. This way the user of the proc macro can define whath is `libparsec_types` should be:
- a mock (the code you were pointing out) in case of `libparsec_serialization_format`'s tests
- the real `libparsec_types` crate for all other cases